asmcomp/cmmgen.ml

```diff
@@ -482,8 +482,10 @@ let rec transl env e =
           | Pbigarray_int32 -> box_int dbg Pint32 elt
           | Pbigarray_int64 -> box_int dbg Pint64 elt
           | Pbigarray_native_int -> box_int dbg Pnativeint elt
-          | Pbigarray_caml_int -> force_tag_int elt dbg
-          | _ -> tag_int elt dbg
+          | Pbigarray_caml_int -> tag_int elt dbg
+          | Pbigarray_sint8 | Pbigarray_uint8
+          | Pbigarray_sint16 | Pbigarray_uint16 -> tag_int elt dbg
+          | Pbigarray_unknown -> assert false
```

Comment:
The case `_caml_int` could be merged with the cases `_sint8`...`_uint16`.  On the other hand, maybe you prefer to have the same case analysis as for "bigarrayset", lines 503-508.

Reply:
Yes, I wanted it to match `bigarrayset`.